### ISSUE TITLE
[Bug]: Unable to set the expiration date when sharing a document

### ISSUE THREAD

**Issue by fgapeters** [nextcloud/server#39310]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

In admin settings a default expiration date for sharing documents is set to 7 days. When users share a document they cannot select an expiration date that is after the currect date plus 7 days. However, one co-worker can set any date in the future for a shared document.
[Nextcloud-Share-ExpirationDate-Bug.docx](https://github.com/nextcloud/server/files/12014729/Nextcloud-Share-ExpirationDate-Bug.docx)

### Steps to reproduce

1. Check admin share settings (in our case 7 days)
2. share a document and note that you cannot change the expiration date more than 7 days ahead.

### Expected behavior

As with older versions the setting of 7 days was a defauly setting. Now it acts as a limit on the maximum number of days that a document can be shared.

### Installation method

Other Community project

### Nextcloud Server version

25

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Nginx

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 22 to 23)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [X] SSO - SAML
- [ ] Other

**Comment by joshtrichards** [comment]
Unclick "Enforce expiration date":

https://docs.nextcloud.com/server/26/admin_manual/configuration_files/file_sharing_configuration.html

**Comment by fgapeters** [comment]
That's not wat I expected and how i was before. The settings screen shows that 7 days is the default value, not the maximum limit.

**Comment by normic** [comment]
@fgapeters problaby has a different expectation, but the description is correct.
We just stumbled about the same issue.

We have Expiration Date enforced and in the past this was the default date set, while creating a share. Now we get the default value set but the date field is read-only. So there is no option to change the Expiration date to a shorter period.

Luckily I found a workaround: One can create the share with the given Expiration date.
Save the share and edit it. Now it's possible to change the Expiration date to a shorter date.
